### PR TITLE
Fixed poi addition to graph

### PR DESCRIPTION
**Pull request by tzufgoogle**
1, Removed assertion such that a POI's projection is on the edge and not beyond it, as might be the case with POI that are not extracted with the graph (e.g., Wikidata items might not be exactly on any street).
2. Changed prefix
3. removed warning of assign in pandas

### CODE REVIEW

File: cabby/geo/walk.py [hunk 1]
```diff
@@ -38,16 +39,13 @@
...
-# This variable is used (:map_structure.py; cabby.geo.walk.py) to project the 
-# geometries into this CRS for geo operations such as calculating the centroid.
-OSM_CRS = 32633  # UTM Zones (North).
 SMALL_POI = 4 # Less than 4 S2Cellids.
 SEED = 4
 MAX_PATH_DIST = 2000
 MIN_PATH_DIST = 200
 NEAR_PIVOT_DIST = 80
 _Geo_DataFrame_Driver = "GPKG"
+MAX_NUM_FAILED = 10
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:48]: max num what failed? (make sure the name indicates this)

File: cabby/geo/walk.py [hunk 2]
```diff
@@ -187,19 +186,22 @@ def get_start_poi(self,
     # Get closest nodes to points.
     dest_osmid = end_point['osmid']
 
-    # Find nodes whithin 2000 meter path distance.
-    outer_circle_graph = ox.truncate.truncate_graph_dist(
-    map.nx_graph, dest_osmid, max_dist=MAX_PATH_DIST, weight='length')
+    try:
+      # Find nodes whithin 2000 meter path distance.
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:190]: sp: whithin -> within

File: cabby/geo/walk.py [hunk 3]
```diff
@@ -627,11 +625,17 @@ def generate_and_save_rvs_routes(self,
...
     while counter < n_samples:
       entity = self.get_single_sample(map)
       if entity is None:
+        attempt += 1
+        if attempt >= MAX_NUM_FAILED:
+          sys.exit("Reached max number of failed attempts.")
         continue
+      attempt = 0 
       counter += 1
+      logging.info("Created sample {}/{}".format(counter,n_samples))
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:638]: It's generally shorter and easier to read when using f'' instead, e.g.:

```
logging.info(f"Created sample {counter}/{n_samples}".)
```

Can you switch to this style here and in future?

**Review by jasonbaldridge — CHANGES_REQUESTED** [review]
The description of the PR is very cryptic. Please write that more clearly and give an example.

A few comments/suggestions in the review to handle, and then this will be all set.

**Review by tzufgoogle — COMMENTED** [review]
1. Changed comments.
2. Added functions for calculating new osmid.
3. Changes string format style.

File: cabby/geo/map_processing/map_structure.py [hunk 4]
```diff
@@ -141,10 +130,13 @@ def add_single_poi_to_graph(
       return single_poi['osmid']
 
     poi_osmid = single_poi['osmid']
-    poi_osmid = util.concat_numbers(999, poi_osmid)
+    poi_osmid = '#' + str(poi_osmid)
+    
+    # If the POI is already in the graph, do not add it.
```
[the remark this answers is not in the document]
Comment (jasonbaldridge) [reply cabby/geo/map_processing/map_structure.py:154]: First, the naming of the function and the semantics are weird. The function is called add_single_poi_to_graph(), which doesn't imply you are going to get any edges back, nor that if the POI exists that nothing happens.

Second, if you are returning a list, then the empty list is a perfect return value when nothing happens. It's hard to reason about null or missing values. You have a narrow use case for handling them, but others may use this same function in a year and not have that context. (

That said, let's move on and I'll consider refactoring it later!

**Review by jasonbaldridge — APPROVED** [review]
LGTM: I'd design some of this differently, but let's get this checked in so you can keep moving.

**Pull request merged**